src/quo2/components/list_items/channel.cljs

```diff
@@ -0,0 +1,57 @@
+(ns quo2.components.list-items.channel
+  (:require [quo.react-native :as rn]
+            [quo2.foundations.colors :as colors]
+            [quo2.components.counter.counter :as quo2.counter]
+            [quo2.components.icon :as quo2.icons]
+            [quo2.components.avatars.channel-avatar :as channel-avatar]
+            [quo2.components.markdown.text :as quo2.text]))
+
+(defn list-item [{:keys [name lock-status mentions-count unread-messages?
+                         muted is-active-channel? emoji channel-color]
+                  :or {channel-color colors/primary-50}}]
+  [rn/view {:style (merge {:height          48
+                           :display         :flex
+                           :border-radius   12
+                           :flex-direction  :row
+                           :justify-content :space-between
+                           :align-items     :center
+                           :width           "100%"
+                           :padding-left    12
+                           :padding-right   12}
+                          (when is-active-channel?
+                            {:background-color (colors/theme-alpha channel-color 0.05 0.05)}))}
+   [rn/view {:display :flex
+             :flex-direction :row
+             :justify-content :flex-start
+             :align-items :center}
+    [channel-avatar/channel-avatar
+     {:big?                   true
+      :lock-status            lock-status
+      :emoji-background-color (colors/theme-alpha channel-color 0.1 0.1)
+      :emoji                  emoji}]
+    [quo2.text/text
+     {:style (merge {:margin-left 12}
+                    (when
+                     (and (not= lock-status :locked)
+                          muted)
+                      {:color colors/neutral-40}))
+      :weight :medium :size :paragraph-1} (str "# " name)]]
+   [rn/view {:style {:height 20}}
+    (when (and (not= lock-status :locked)
+               muted)
+      [quo2.icons/icon :main-icons2/muted
+       {:size     20
+        :no-color true}])
+    (when (and (not= lock-status :locked)
+               (not muted)
+               (pos? (int mentions-count)))
+      [rn/view {:style {:margin-right 2
+                        :margin-top   2}}
+       [quo2.counter/counter {:override-bg-color channel-color} mentions-count]])
+    (when (and (not= lock-status :locked)
```

Comment:
If you were interested in using a boolean `locked?` instead of a map `lock-status`, maybe you could test the values like this:

```clojure
(def locked? nil) ; false | true | nil
(println "[locked?]" locked?
          "[unlocked?]" (or (nil? locked?) (false? locked?))
          "[lock exists?]" (some? locked?)
          "[unlocked or lock doesn't exist]" (or (false? locked?) (nil? locked?)))
```

I can see the simplicity of just having 3 values in a `lock-status` map, though. On the other hand, I kinda like seeing `locked?` in the code instead of `lock-status`.

Reply:
@ibrkhalil, @Parveshdhull WDYT? If I make this change will also adjust in the avatar-channel component too as I was just matching the api here.

Reply:
>  "[unlocked or lock doesn't exist]" (or (false? locked?) (nil? locked?)))

This can get little confusing, as having nil value of locked can mean both, unlocked or lock doesn't exist.

I guess if want to use boolean, we can create two variables (as we already do that later in main component [ref](https://github.com/status-im/status-mobile/blob/147958dd135aeb5abbf1c792e70f8fcfcc2c4c91/src/quo2/components/avatars/channel_avatar.cljs#L10))
1. locked?
2. show-lock-icon?

Reply:
This will require changes in `Channel-avatar` of course we can have two values and check for `nil` on lockekd?
I second Erik's suggestion

Reply:
Updated to include these changes into both list item channel and channel avatar component